kernel/names.ml

```diff
@@ -917,11 +917,13 @@ struct
       if npars == npars' then p
       else {p with proj_npars = npars'}
 
-    let map f p =
+    let map f_ind f_const p =
```

Comment:
Alternatively don't take a f_const here and instead rebuild the constant when `ind' != ind`

Reply:
also we could expose a Projection.canonize instead of using `map`

Reply:
That seems easier and less invasive than my change.

Reply:
Github is not loading new comments automatically for some reason so I only saw your second comment after I responded to the first. I am not sure how `canonize` will help with substitution. Is the idea to recompute the constant in `canonize`?

Reply:
I see that half your changes are about code trying to canonize, so I think we should expose that as an API instead of reimplementing it everywhere.

Reply:
call it a mildly related cleanup if you want

Reply:
I think that type of cleanup probably requires deeper understand of the code that I do not have. I was very happy to come up with the changes I have here and I could probably implement the solution that recomputes the name and doesn't change the signature of `map` but I don't think I can easily clean up the code that is there already.

Reply:
I implemented the proposal of recomputing the projection.